libvirt/tests/src/virtual_network/iface_network.py

```
@@ -560,6 +563,22 @@ def run_guest_libvirt(session):
         if not utils_package.package_remove("libvirt*", session):
             test.error("Failed to remove libvirt packages on guest")
 
+    def check_qdisc(expect_type):
+        """
+        check the qdisc type of the tap device
+
+        param expect_type: the expected qdisc type for the interface, for example, noqueue, mq, htb or others.
+        """
+        if not vm.is_alive():
+            vm.start()
+        mac = vm_xml.VMXML.get_iface_dev(vm_name)[0]
+        iface_features = vm_xml.VMXML.get_iface_by_mac(vm_name, mac)
+        target_dev = iface_features.get('target', {}).get('dev')
+        if target_dev:
+            outputs = process.run("ip l show %s" % target_dev).stdout_text
+        if not re.findall(expect_type, outputs):
+            test.fail("The qdisc type should be %s, but it is not, check %s" % (expect_type, outputs))
```

outputs will not be defined if target_dev is None, right? Should we raise an exception in this case?

---

Add a check before the line, and update it to be:
        if not target_dev:
            test.skip("Failed to get the target dev name for qdisc test!")
        outputs = process.run("ip l show %s" % target_dev).stdout_text